core/src/main/java/kafka/server/TierStateMachine.java

```diff
@@ -229,12 +229,8 @@ private Long buildRemoteLogAuxState(TopicPartition topicPartition,
         }
 
         RemoteLogSegmentMetadata remoteLogSegmentMetadata = rlm.fetchRemoteLogSegmentMetadata(topicPartition, targetEpoch, previousOffsetToLeaderLocalLogStartOffset)
-                .orElseThrow(() -> new RemoteStorageException("Couldn't build the state from remote store for partition: " + topicPartition +
-                        ", currentLeaderEpoch: " + currentLeaderEpoch +
-                        ", leaderLocalLogStartOffset: " + leaderLocalLogStartOffset +
-                        ", leaderLogStartOffset: " + leaderLogStartOffset +
-                        ", epoch: " + targetEpoch +
-                        "as the previous remote log segment metadata was not found"));
+                .orElseThrow(() -> buildRemoteStorageException(topicPartition, targetEpoch, currentLeaderEpoch,
+                        leaderLocalLogStartOffset, previousOffsetToLeaderLocalLogStartOffset));
```

Comment:
It seems to me that the final argument here should be `leaderLogStartOffset` to match the previous behaviour.

Reply:
Thx. Addressed in the recent commit.

Reply:
@AndrewJSchofield Additional ping for possible notification miss